core/atrium-core-common/src/main/kotlin/ch/tutteli/atrium/assertions/builders/descriptiveWithFailureHint.kt

```diff
@@ -27,20 +40,50 @@ fun Descriptive.DescriptionOption<Descriptive.FinalStep>.withFailureHint(
  * which is based on the subject of the expectation and
  * which is only shown the subject is defined.
  *
- * You can use [withFailureHintBasedOnSubject] in case you want to:
+ * You can use [withHelpOnFailureBasedOnSubject] in case you want to:
  * - provide a hint also if the subject is absent.
  * - do not show the hint in certain cases even if the subject is defined
  *
- * Or use [withFailureHint] which does not expect a [subjectProvider] in case your [DescriptiveAssertion] is not based
+ * Or use [withHelpOnFailure] which does not expect a [subjectProvider] in case your [DescriptiveAssertion] is not based
  * on the subject of the expectation.
  */
 //TODO if we introduce Record or something else as replacement for Assertion then not but if we keep Assertion
 // then move to logic and expect ProofContainer with 0.18.0
+fun <T> Descriptive.DescriptionOption<Descriptive.FinalStep>.withHelpOnFailureBasedOnDefinedSubject(
+    expect: Expect<T>,
+    failureHintFactory: (T) -> Assertion
+): Descriptive.DescriptionOption<DescriptiveAssertionWithFailureHint.FinalStep> {
+    return withHelpOnFailureBasedOnSubject(expect) {
+        ifDefined(failureHintFactory)
+            .ifAbsent {
+                assertionBuilder.explanatoryGroup
+                    .withWarningType
+                    .withExplanatoryAssertion(Text(SHOULD_NOT_BE_SHOWN_TO_THE_USER_BUG))
+                    .build()
+            }
+    }.showOnlyIfSubjectDefined(expect)
+}
+
+/**
+ * Option to create a [DescriptiveAssertion] like assertion with an additional hint
+ * which is based on the subject of the expectation and
+ * which is only shown the subject is defined.
+ *
+ * You can use [withHelpOnFailureBasedOnSubject] in case you want to:
+ * - provide a hint also if the subject is absent.
+ * - do not show the hint in certain cases even if the subject is defined
+ *
+ * Or use [withHelpOnFailure] which does not expect a [subjectProvider] in case your [DescriptiveAssertion] is not based
+ * on the subject of the expectation.
+ */
+//TODO if we introduce Record or something else as replacement for Assertion then not but if we keep Assertion
+// then move to logic and expect ProofContainer with 0.18.0
+@Deprecated("Use withHelpOnFailureBasedOnDefinedSubject; will be removed with 1.0.0")
 fun <T> Descriptive.DescriptionOption<Descriptive.FinalStep>.withFailureHintBasedOnDefinedSubject(
     expect: Expect<T>,
     failureHintFactory: (T) -> Assertion
 ): Descriptive.DescriptionOption<DescriptiveAssertionWithFailureHint.FinalStep> {
-    return withFailureHintBasedOnSubject(expect) {
+    return withHelpOnFailureBasedOnSubject(expect) {
```

Comment:
I would delegate to `withHelpOnFailureBasedOnDefinedSubject` but also fine like this

Reply:
Hey @robstoll! I can change that if you would like for me to do so - I was just flipping any old `with*failureHint` method names to the new equivalent on all of the usages I could find to make sure I didn't introduce any unexpected changes. I'll adjust this on your signal.